packages/block-library/src/site-logo/index.php

```diff
@@ -111,24 +111,26 @@ function _override_custom_logo_theme_mod( $custom_logo ) {
 /**
  * Updates the site_logo option when the custom_logo theme-mod gets updated.
  *
- * @param string $custom_logo The custom logo set by a theme.
+ * This function is hooked on "update_option_theme_mods_$theme" and not
+ * "pre_set_theme_mod_custom_logo" because by hooking in `update_option`
+ * the function accounts for remove_theme_mod() as well.
  *
- * @return string The custom logo.
+ * @param mixed $old_value The old option value.
+ * @param mixed $value     The new option value.
  */
-function _sync_custom_logo_to_site_logo( $custom_logo ) {
+function _sync_custom_logo_to_site_logo( $old_value, $value ) {
 	// Delete the option when the custom logo does not exist or was removed.
 	// This step ensures the option stays in sync.
-	if ( empty( $custom_logo ) ) {
+	if ( empty( $value['custom_logo'] ) ) {
 		delete_option( 'site_logo' );
 	} else {
 		remove_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo' );
-		update_option( 'site_logo', $custom_logo );
+		update_option( 'site_logo', $value['custom_logo'] );
 		add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
 	}
-	return $custom_logo;
 }
 
-add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+add_action( 'update_option_theme_mods_' . get_option( 'stylesheet' ), '_sync_custom_logo_to_site_logo', 10, 2 );
```

Comment:
The only other thing that comes to mind is it may be preferable to add this action later. This would ensure that if the custom logo is changed when a dynamically-switched theme is active, it would also get synchronized. For example, the customizer switches the theme at `setup_theme` priority 10:

https://github.com/WordPress/wordpress-develop/blob/e37b85fcd7b49846679caef7701c89fcd5c22f0f/src/wp-includes/class-wp-customize-manager.php#L675-L687

So to be sure to catch this case, you may want to do:

```suggestion
add_action(
	'setup_theme',
	static function () {
		add_action( 'update_option_theme_mods_' . get_option( 'stylesheet' ), '_sync_custom_logo_to_site_logo', 10, 2 );
	},
	11
);
```

Reply:
Thank you Weston! Fixed :+1: